tests/issues.rs

```diff
@@ -166,3 +166,16 @@ fn test_issue1() {
         ]
     );
 }
+
+#[test]
+fn test_pr12() {
+    struct Recv {}
+    impl SpannedEventReceiver for Recv {
+        fn on_event(&mut self, _ev: Event, _span: saphyr_parser::Span) {}
+    }
+    let s = "---\n- |\n  a";
+    let input = BufferedInput::new(s.chars());
+    let mut parser = Parser::new(input);
+    let mut recv = Recv {};
+    parser.load(&mut recv, true).unwrap();
+}
```

Comment:
Can this be re-written using the events directly? This should be

```rs
#[test]
fn test_pr12() {
    assert_eq!(
        run_parser("---\n- |\n  a").unwrap(),
        [
            Event::StreamStart,
            Event::DocumentStart(true),
            Event::SequenceStart(0, None),
            Event::Scalar("a\n".to_string(), TScalarStyle::Literal, 0, None),
            Event::SequenceEnd,
            Event::DocumentEnd,
            Event::StreamEnd,
        ]
    );
}
```

Reply:
Sure, although `run_parser` doesn't test the right code path since its a `StrInput`. I added a `run_parser_buffered` function.